sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,502 @@
+  /**
+   * The base address for the Pin Multiplexer hardware registers.
+   */
+  mmio_region_t base_addr;
+} dif_pinmux_params_t;
+
+/**
+ * Pinmux Padring pin kinds.
+ *
+ * A pin can be a Multiplexed (DIO) or Dedicated (DIO).
+ */
+typedef enum dif_pinmux_padring_pin_kind {
+  /**
+   * Multiplexed Input/Output pin. Connected to a peripheral input or output
+   * via a multiplexer matrix inside the Pin Multiplexer.
+   */
+  kDifPinmuxPadringPinKindMio = 0,
+  /**
+   * Dedicated Input/Output pin. Connected directly to a peripheral input
+   * or output, bypassing the multiplexer matrix.
+   */
+  kDifPinmuxPadringPinKindDio,
+} dif_pinmux_padring_pin_kind_t;
+
+/**
+ * Pinmux functionality locking.
+ *
+ * Represents a type of functionality that can be locked inside the
+ * Pin Multiplexer for a given pin/peripheral input.
+ */
+typedef enum dif_pinmux_lock_type {
+  /**
+   * Lock Padring MIO input select for a given peripheral input. This means that
+   * the peripheral input is connection can no longer be changed.
+   */
+  kDifPinmuxLockTypeInsel = 0,
+  /**
+   * Lock peripheral output select for a given Padring MIO output. This means
+   * that the Padring MIO output connection can no longer be changed.
+   */
+  kDifPinmuxLockTypeOutsel,
+  /**
+   * Lock the sleep mode configuration for a given Padring MIO pin.
+   */
+  kDifPinmuxLockTypeMioSleep,
+  /**
+   * Lock the sleep mode configuration for a given Padring DIO pin.
+   */
+  kDifPinmuxLockTypeDioSleep,
+  /**
+   * Lock physical characteristics configuration for a given Padring MIO pin.
+   */
+  kDifPinmuxLockTypeMioPinAttr,
+  /**
+   * Lock physical characteristics configuration for a given Padring DIO pin.
+   */
+  kDifPinmuxLockTypeDioPinAttr,
+} dif_pinmux_lock_type_t;
+
+/**
+ * Pinmux generic index.
+ *
+ * The meaning of this index changes between APIs, and must be documented in
+ * the corresponding function description.
+ */
+typedef uint32_t dif_pinmux_index_t;
+
+/**
+ * Multiplexed Padring input connection select for a peripheral input.
+ */
+typedef uint32_t dif_pinmux_padring_insel_t;
+
+/**
+ * Multiplexed peripheral output connection select for a Padring output.
+ */
+typedef uint32_t dif_pinmux_peripheral_outsel_t;
+
+/**
+ * Pinmux Padring pin attributes.
+ *
+ * Some attributes have defined meanings, others are implementation defined. Not
+ * all pins will support all attributes. The Pinmux DIF does not check for
+ * unsupported attributes. To check to see if an attribute is supported after
+ * enabling it (via `dif_pinmux_padring_attr_enable` or
+ * `dif_pinmux_padring_attr_write`) read back the attribute field (via
+ * `dif_pinmux_padring_attr_get`). If the attribute isn't supported the
+ * attribute will still be disabled (relevant bit unset in the attribute field).
+ */
+typedef enum dif_pinmux_padring_pin_attr {
+  kDifPinmuxPadringPinAttrNone = 0,
+  kDifPinmuxPadringPinAttrIOInvert = 1 << 0,
+  kDifPinmuxPadringPinAttrOpenDrain = 1 << 1,
```

Comment:
It would be good if we called this `VirtOpenDrain`, since it is an emulated open drain mode and not a "real" open drain mode (from an electrical standpoint).

I.e., In this virtual open drain mode, the pads are put into high-z when driving out a logic 1, whereas logic 0's just drive out an electrical logic low level.